my_game.py

```diff
@@ -147,26 +181,18 @@ def on_show_view(self):
 
         self.player_score = 0
 
-        self.player_sprite_list = []
+        # self.player_sprite_list = []
 
         # replace all sprites on layer "players" with actual player objects
-        for i in range(NUM_OF_PLAYERS):
+
+        for i in range(len(self.player_sprite_list)):
             # Creates Player object
-            p = Player(
-                position=(self.tilemap.sprite_lists["players"][0].center_x + random.randint(1,8) * TILE_SIZE * SCALING, self.tilemap.sprite_lists["players"][0].center_y),
-                max_hp=20,  # FIXME: add some kind of config for the player to avoid magic numbers
-                speed=3,
-                window=self.window,
-                equipped_weapon=Weapon(type=WeaponType.SWORD_SHORT),
-                scale=SCALING,
-                key_up=PLAYER_KEYS[i]["up"],
-                key_down=PLAYER_KEYS[i]["down"],
-                key_left=PLAYER_KEYS[i]["left"],
-                key_right=PLAYER_KEYS[i]["right"],
-                key_attack=PLAYER_KEYS[i]["attack"],
+
+            self.player_sprite_list[i].position = (
+                self.tilemap.sprite_lists["players"][0].center_x + random.randint(1,8) * TILE_SIZE * SCALING,
+                self.tilemap.sprite_lists["players"][0].center_y
```

Comment:
We should spawn at the dedicated tiles. We have a tile for each player, right?

Reply:
Yes we do!